pkg/k8s/watchers/pod.go

```diff
@@ -194,7 +193,19 @@ func (k *K8sWatcher) addK8sPodV1(pod *slim_corev1.Pod) error {
 		"hostIP":               pod.Status.PodIP,
 	})
 
-	skipped, err := k.updatePodHostData(pod)
+	skipped := false
+	podIPs, err := k8sUtils.ValidIPs(pod.Status)
+
+	if err == nil {
+		skipped, err = k.updatePodHostData(pod, podIPs)
+
+		// There might be duplicate callbacks here since this function is also
+		// called from updateK8sPodV1, the consumer will need to handle the duplicate
+		// events accordingly.
+		// GH issue #13136.
+		k.redirectPolicyManager.OnAddPod(pod)
```

Comment:
I've added a check in all the redirect policy manager to return if there are no LRPs. As for the cases where there are LRPs, we can consider this making this asynchronous. However, I do see other code paths in the existing code that are blocking [1]. It can be a bigger refactoring change. 

[1] https://github.com/cilium/cilium/blob/master/pkg/k8s/watchers/pod.go#L197